lawnchair/src/app/lawnchair/nexuslauncher/OverlayCallbackImpl.kt

```diff
@@ -0,0 +1,136 @@
+package app.lawnchair.nexuslauncher
+
+import android.app.Activity
+import android.content.Context
+import android.content.pm.ApplicationInfo
+import android.os.Bundle
+import app.lawnchair.FeedBridge
+import app.lawnchair.LawnchairLauncher
+import app.lawnchair.preferences2.PreferenceManager2
+import com.android.launcher3.Utilities
+import com.android.systemui.plugins.shared.LauncherOverlayManager
+import com.android.systemui.plugins.shared.LauncherOverlayManager.LauncherOverlay
+import com.android.systemui.plugins.shared.LauncherOverlayManager.LauncherOverlayCallbacks
+import com.google.android.libraries.launcherclient.ISerializableScrollCallback
+import com.google.android.libraries.launcherclient.LauncherClient
+import com.google.android.libraries.launcherclient.LauncherClientCallbacks
+import com.google.android.libraries.launcherclient.LauncherClientService
+import com.google.android.libraries.launcherclient.StaticInteger
+import com.patrykmichalik.opto.core.firstBlocking
+
+/**
+ * Implements [LauncherOverlay] and passes all the corresponding events to [LauncherClient],
+ * see [LauncherClientService.setClient].
+ *
+ * Implements [LauncherClientCallbacks] and sends all the corresponding callbacks to [ ].
+ */
+class OverlayCallbackImpl(private val mLauncher: LawnchairLauncher) : LauncherOverlay,
+    LauncherClientCallbacks, LauncherOverlayManager, ISerializableScrollCallback {
+    private val mClient: LauncherClient
+    private var mFlagsChanged = false
+    private var mLauncherOverlayCallbacks: LauncherOverlayCallbacks? = null
+    private var mWasOverlayAttached = false
+    private var mFlags = 0
+
+    init {
+        val enableFeed = PreferenceManager2.getInstance(mLauncher).enableFeed.firstBlocking()
+        mClient = LauncherClient(
+            mLauncher, this, StaticInteger((if (enableFeed) 1 else 0) or 2 or 4 or 8)
+        )
+    }
+
+    override fun onDeviceProvideChanged() {
+        mClient.redraw()
+    }
+
+    override fun onAttachedToWindow() {
+        mClient.onAttachedToWindow()
+    }
+
+    override fun onDetachedFromWindow() {
+        mClient.onDetachedFromWindow()
+    }
+
+    override fun openOverlay() {
+        mClient.showOverlay(true)
+    }
+
+    override fun hideOverlay(animate: Boolean) {
+        mClient.hideOverlay(animate)
+    }
+
+    override fun hideOverlay(duration: Int) {
+        mClient.hideOverlay(duration)
+    }
+
```

Comment:
```suggestion

    override fun startSearch(config: ByteArray?, extras: Bundle?): Boolean {
        return false
    }

    override fun onActivityCreated(activity: Activity, bundle: Bundle?) {
        // Not called
    }

```

These two overrides are missing from the new Kotlin file, I know they match the super methods but as far as I know, they should be still overridden here.

Reply:
be1bdf847d6593257490dc28d8c63f9dfcf36b87

Reply:
Nice catch!